drivers/SmartThings/philips-hue/src/disco/button.lua

```diff
@@ -0,0 +1,148 @@
+
+      if control_id == 1 and button_remote_description.id == nil then
+        button_remote_description.id = button_repr.data[1].id
+      end
+    end
+  end
+
+  local battery, battery_err = api_instance:get_device_power_by_id(rid_by_rtype[HueDeviceTypes.DEVICE_POWER])
+  if battery_err then return nil, battery_err end
+
+  if battery and battery.data and battery.data[1] then
+    button_remote_description.power_id = battery.data[1].id
+    button_remote_description.power_state = battery.data[1].power_state
+  end
+
+  if type(cache) == "table" then
+    cache[resource_id] = button_remote_description
+    if device_service_info.id_v1 then
+      cache[device_service_info.id_v1] = button_remote_description
+    end
+  end
+
+  return button_remote_description
+end
+
+---@param api_instance PhilipsHueApi
+---@param device_service_id string
+---@param bridge_id string
+---@param primary_button_resource_id string
+---@param cache table?
+---@return table<string,any>? description nil on error
+---@return string? err nil on success
+function M.update_state_for_all_device_services(api_instance, device_service_id, bridge_id, primary_button_resource_id, cache)
+  log.debug("----------- Calling REST API")
+  local device_service_info, err = api_instance:get_device_by_id(device_service_id)
+  if err or not (device_service_info and device_service_info.data) then
+    log.error("Couldn't get device info for button, error: " .. st_utils.stringify_table(err))
+    return
+  end
+
+  log.debug("------------ _do_update")
+  return _do_update(api_instance, device_service_info.data[1], bridge_id, primary_button_resource_id, cache)
+end
+
+---@param driver HueDriver
+---@param bridge_id string
+---@param api_instance PhilipsHueApi
+---@param resource_id string
+---@param device_service_info HueDeviceInfo
+---@param device_state_disco_cache table<string, table>
+---@param st_metadata_callback fun(driver: HueDriver, metadata: table)?
+function M.handle_discovered_device(
+    driver, bridge_id, api_instance,
+    resource_id, device_service_info,
+    device_state_disco_cache, st_metadata_callback
+)
+  local button_description, err = _do_update(
+    api_instance, device_service_info, bridge_id, resource_id, device_state_disco_cache
+  )
+  if err then
+    log.error("Error updating contact button initial state: " .. st_utils.stringify_table(err))
+    return
+  end
+
+  if type(st_metadata_callback) == "function" then
+    if not (button_description and HueDeviceTypes.supports_button_configuration(button_description)) then
+      button_description = button_description or {num_buttons = "unknown"}
+      log.error(
+        string.format(
+          "Driver does not currently support remotes with %s buttons, cannot create device", button_description.num_buttons
+        )
+      )
+      return
+    end
+    local button_profile_ref = ""
+    if button_description.num_buttons == 4 then
+      button_profile_ref = "4-button-remote"
```

Comment:
I think we probably want to log the error case here and skip the device creation since without a profile the device isn't going to be useable. Thoughts?

Reply:
Agreed, good call out.

Reply:
Fixed on https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1359